inference-engine/src/multi_device/multi_device_exec_network.hpp

```diff
@@ -56,15 +57,40 @@ class ThreadSafeQueue {
             return false;
         }
     }
-
-    bool empty() {
+protected:
+    std::queue<T>   _queue;
+    std::mutex      _mutex;
+};
+template <typename T>
+class ThreadSafeBoundedQueue {
+public:
+    ThreadSafeBoundedQueue() = default;
+    bool try_push(T value) {
+        std::lock_guard<std::mutex> lock(_mutex);
+        if (_capacity) {
```

Comment:
Shall you check that the capacity is >= than current size+1?

Reply:
No